core/metrics_manager.js

```diff
@@ -0,0 +1,385 @@
+ *     contents of the given workspace, as an object containing
+ *     - height and width in pixels
+ *     - left and top in pixels relative to the workspace origin.
+ * @protected
+ */
+Blockly.MetricsManager.prototype.getContentDimensionsBounded_ = function(
+    viewMetrics) {
+  var content = this.getContentDimensionsExact_();
+  var contentRight = content.left + content.width;
+  var contentBottom = content.top + content.height;
+
+  // View height and width are both in pixels, and are the same as the SVG size.
+  var viewWidth = viewMetrics.width;
+  var viewHeight = viewMetrics.height;
+  var halfWidth = viewWidth / 2;
+  var halfHeight = viewHeight / 2;
+
+  // Add a border around the content that is at least half a screen wide.
+  // Ensure border is wide enough that blocks can scroll over entire screen.
+  var left = Math.min(content.left - halfWidth, contentRight - viewWidth);
+  var right = Math.max(contentRight + halfWidth, content.left + viewWidth);
+
+  var top = Math.min(content.top - halfHeight, contentBottom - viewHeight);
+  var bottom = Math.max(contentBottom + halfHeight, content.top + viewHeight);
+
+  return {left: left, top: top, height: bottom - top, width: right - left};
+};
+
+/**
+ * Gets the bounding box for all workspace contents, in pixel coordinates.
+ * @return {!Blockly.MetricsManager.ContainerRegion} The dimensions of the
+ *     contents of the given workspace in pixel coordinates, as an object
+ *     containing
+ *     - height and width in pixels
+ *     - left and top in pixels relative to the workspace origin.
+ * @protected
+ */
+Blockly.MetricsManager.prototype.getContentDimensionsExact_ = function() {
+  // Block bounding box is in workspace coordinates.
+  var blockBox = this.workspace_.getBlocksBoundingBox();
+  var scale = this.workspace_.scale;
+
+  // Convert to pixels.
+  var top = blockBox.top * scale;
+  var bottom = blockBox.bottom * scale;
+  var left = blockBox.left * scale;
+  var right = blockBox.right * scale;
+
+  return {top: top, left: left, width: right - left, height: bottom - top};
+};
+
+/**
+ * Gets the width and the height of the flyout on the workspace in pixel
+ * coordinates. Returns 0 for the width and height if the workspace has a
+ * category toolbox instead of a simple toolbox.
+ * @param {boolean=} opt_own Only return the workspace's own flyout if True.
+ * @return {!Blockly.utils.Size} The width and height of the flyout.
+ * @public
+ */
+Blockly.MetricsManager.prototype.getFlyoutMetrics = function(opt_own) {
+  var flyoutDimensions = this.getDimensionsPx_(this.workspace_.getFlyout(opt_own));
+  return new Blockly.utils.Size(
+      flyoutDimensions.width, flyoutDimensions.height);
+};
+
+/**
+ * Gets the width, height and position of the toolbox on the workspace in pixel
+ * coordinates. Returns 0 for the width and height if the workspace has a simple
+ * toolbox instead of a category toolbox. To get the width and height of a
+ * simple toolbox see getFlyoutMetrics.
```

Comment:
I think there's an annotation you can use like `@see` and/or `@link` that you can use for this reference to `getFlyoutMetrics`

Reply:
Ok, I added it but couldn't find a lot of information on how it shows up in the docs so I am not sure if I did it correctly. Let me know if you think it should be changed.